cpp/include/cuspatial/spatial_join.hpp

```diff
@@ -51,44 +52,97 @@ namespace cuspatial {
-  rmm::mr::device_memory_resource *mr = rmm::mr::get_default_resource());
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
 
 /**
- * @brief pair points and polygons using ray-cast based point-in-polygon test algorithm in two
- *phases: phase 1 counts the total number of output pairs for precise memory allocation phase 2
- *actually writes (point,polygon) pairs
- *
- * @param pq_pair: table of two arrays for (quadrant,polygon) pairs derived from spatial filtering
- * @param quadtree: table of five arrays derived from quadtree indexing on points (key,lev,sign,
- * length, fpos)
- * @param pnt: table of two arrays for points (x,y). note that points are in-place sorted in
- * quadtree construction and have different orders than the orginal input points.
- * @param fpos: feature/polygon offset array to rings
- * @param rpos: ring offset array to vertex
- * @param poly_x: polygon x coordiante array.
- * @param poly_y: polygon y coordiante array.
- *
- * @return array of (polygon_index, point_index) pairs that point is within polyon;
- * point_index and polygon_index are offsets of point and polygon arrays, respectively
- */
-// std::unique_ptr<cudf::table> pip_refine(
-//   cudf::table_view const& pq_pair,
-//   cudf::table_view const& quadtree,
-//   cudf::table_view const& pnt,
-//   cudf::column_view const& poly_fpos,
-//   cudf::column_view const& poly_rpos,
-//   cudf::column_view const& poly_x,
-//   cudf::column_view const& poly_y,
-//   rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
+ * @brief Finds points in a set of (polygon, quadrant) pairs derived from spatial filtering.
+ *
+ * @param poly_quad_pairs Table of (quadrant, polygon) index pairs derived from spatial filtering.
+ * @param quadtree Table representing a quadtree (key, level, is_quad, length, offset).
+ * @param point_indices Sorted indices of quadtree points
+ * @param point_x x-coordinates of points to test
+ * @param point_y y-coordinates of points to test
+ * @param poly_offsets Begin indices of the first ring in each polygon (i.e. prefix-sum).
+ * @param ring_offsets Begin indices of the first point in each ring (i.e. prefix-sum).
+ * @param poly_points_x Polygon point x-coodinates.
+ * @param poly_points_y Polygon point y-coodinates.
+ * @param mr The optional resource to use for output device memory allocations.
+ *
+ * @throw cuspatial::logic_error If the poly_quad_pairs table is malformed.
+ * @throw cuspatial::logic_error If the quadtree table is malformed.
+ * @throw cuspatial::logic_error If the number of point indices doesn't match the number of points.
+ * @throw cuspatial::logic_error If the number of rings is less than the number of polygons.
+ * @throw cuspatial::logic_error If each ring has fewer than three vertices.
+ * @throw cuspatial::logic_error If the types of point and polygon vertices are different.
+ *
+ * @returns A cudf table with two columns of point/polygon pairs for each intersection:
```

Comment:
Are there two columns per intersection or one pair per intersection? Clarify.

Reply:
One pair per intersection, but I can see how the wording can be better :pray: